core/src/test/java/com/google/cloud/sql/core/CoreSocketFactoryTest.java

```diff
@@ -347,31 +346,119 @@ public void create_notAuthorizedToGetInstance() throws IOException {
 
   @Test
   public void create_notAuthorizedToCreateEphemeralCertificate() throws IOException {
-    ErrorInfo error = new ErrorInfo();
-    error.setReason(CoreSocketFactory.INSTANCE_NOT_AUTHORIZED_REASON);
-    GoogleJsonError details = new GoogleJsonError();
-    details.setErrors(ImmutableList.of(error));
-    when(adminApiSslCertsCreateEphemeral.execute())
-        .thenThrow(
-            new GoogleJsonResponseException(
-                new HttpResponseException.Builder(403, "Forbidden", new HttpHeaders()), details));
-
     CoreSocketFactory coreSocketFactory =
         new CoreSocketFactory(clientKeyPair, credential, adminApi, 3307, defaultExecutor);
     try {
-      coreSocketFactory.createSslSocket(INSTANCE_CONNECTION_STRING, Arrays.asList("PRIMARY"));
+      // Use a different instance to simulate incorrect permissions.
+      coreSocketFactory.createSslSocket(
+          "myProject:myRegion:NotMyInstance", Arrays.asList("PRIMARY"));
       fail();
     } catch (RuntimeException e) {
-      // TODO(berezv): should we throw something more specific than RuntimeException?
       assertThat(e)
           .hasMessageThat()
           .contains(
               String.format(
                   "[%s] The Cloud SQL Instance does not exist or your account is not authorized",
-                  INSTANCE_CONNECTION_STRING));
+                  "myProject:myRegion:NotMyInstance"));
     }
   }
 
+  /*
+   * Creates a fake "accessNotConfigured" exception that can be used for testing.
+   *
+   * Example Response:
+   *  {
```

Comment:
The code relies on the contents of the "reason" field - I left the errors in the comments because I didn't want to have to hunt for them in the future, but I'll move then into a personal doc.

Reply:
OK, that's fine. But if it doesn't rely on the "message" field, perhaps you don't need to put all that stuff. "message goes here" might be enough.